openml/tasks/functions.py

```diff
@@ -236,7 +236,7 @@ def get_task(task_id):
 
     # TODO look into either adding the class labels to task xml, or other
     # way of reading it.
-    class_labels = dataset._retrieve_class_labels()
+    class_labels = dataset._retrieve_class_labels(task.target_feature)
```

Comment:
I don't understand this. What is this for regression or clustering or whatever? Is this classification specific? Is this None otherwise?
Also, as I said in my other comment, ``target_feature`` is a pretty bad name. I think it's called ``target`` in other places but maybe it should be ``target_name``? We should definitely unify this.